diff --git a/duo_workflow_service/agents/chat_agent.py b/duo_workflow_service/agents/chat_agent.py
index 13128a1c1..510742850 100644
--- a/duo_workflow_service/agents/chat_agent.py
+++ b/duo_workflow_service/agents/chat_agent.py
@@ -6,6 +6,7 @@ import structlog
 from anthropic import APIStatusError
 from langchain_core.messages import AIMessage, BaseMessage, HumanMessage, ToolMessage
 
+from ai_gateway.app import get_config
 from duo_workflow_service.agents.prompt_adapter import BasePromptAdapter
 from duo_workflow_service.components.tools_registry import ToolsRegistry
 from duo_workflow_service.entities.state import (
@@ -48,7 +49,12 @@ class ChatAgent:
         self, message: AIMessage, preapproved_tools: List[str]
     ) -> tuple[bool, list[UiChatLog]]:
         approval_required = False
-        approval_messages = []
+        approval_messages: list[UiChatLog] = []
+
+        # If caching proxy is enabled don't require approvals (they interrupt the automated tests) the proxy facilitates
+        config = get_config()
+        if config.duo_workflow.use_caching_proxy:
+            return approval_required, approval_messages
 
         for call in message.tool_calls:
             tool_name = call["name"]
